docs/advanced-topics/advanced-methods.md

```diff
@@ -0,0 +1,100 @@
+and attempts to minimize the *angular distance* between the document and its class centroid, and maximise the angular distance between the document and the centroids of the other classes.
+
+![training](../imgs/SphereFace-training.png)
+
+The ArcFace and CosFace both deviate from the traditional SphereFace by including a margin and scaling parameter, which can be used to increase the boundary between each class. If an item's embedding is within the boundary of the class it belongs to, then no loss is incurred. Choosing appropriate values for the margin and scaling parameter is important for effective training, for more information on how ArcFace and CosFace calculate loss, and how these parameters affect the output, see this article on [loss metrics for deep learning](https://hav4ik.github.io/articles/deep-metric-learning-survey#cosface).  
+
+Since only one sample is needed to calculate the loss with the ArcFace and CosFace functions, there are no constraints on what each batch needs to contain, unlike Triplet Margin Loss
+which uses a `ClassSampler` to construct batches an equal amount of each class in the batch.
+Instead, we can construct batches using random sampling, a much simpler method which consequently takes less time to construct a batch.
+By default, runs created using ArcFace or Cosface loss will use random sampling, however you can specify which type of sampling method you would like to use like so:
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   sampler='random'      # use random sampling
++   sampler='class'       # use class sampling
++   sampler='auto'        # infer sampling method based on the loss function (default)
+
+)
+```
+
+Since Triplet Margin Loss needs a class sampler to properly function, this `sampler` parameter is ignored when `loss=TripletMarginLoss`.
+
+### Using an Optimizer
+
+In order to keep track of the class centers across batches, these SphereFace loss functions require an additional optimizer during training.
+By default, the type of optimizer used will be the same as the one used for the model itself, but you can also choose a different optimizer for your loss function using the `loss_optimizer` parameter.
+The list of available optimizers are discussed in the [run job](../walkthrough/run-job.md) section.
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   loss_optimizer='Adam',
++   loss_optimizer_options={'weight_decay': 0.01}
+)
+```
+
+Using these loss functions over the default Triplet Margin Loss can result in clearer divisions between the domains representing each class in the embedding space.
+As an example, the figure below shows the domains of the 10 classes of the FMNIST dataset projected onto 2D space after training with Triplet Loss, ArcFace Loss and CosFace Loss.
```

Comment:
let's add a reference to the dataset, also let's mention the embeddings is projected into 2d using umap